airbyte-cdk/python/airbyte_cdk/connector_builder/message_grouper.py

```diff
@@ -45,6 +45,18 @@ def __init__(self, max_pages_per_slice: int, max_slices: int, max_record_limit:
         self._max_slices = max_slices
         self._max_record_limit = max_record_limit
 
+    def _to_nested_and_composite_field(self, field: Optional[Union[str, List[str], List[List[str]]]]) -> List[List[str]]:
```

Comment:
This hasn't been tested. It is a bit annoying to have to bother so much about the shape of a field. If you think this is necessary, I can add those tests. Right now, I've only tested manually using the example in the PR description and PK `["rates", "AED"]`